test/network/scenarios/p2p/peer_disconnect.js

```diff
@@ -54,10 +54,14 @@ module.exports = function(
 
 			describe('when a node is stopped', () => {
 				before(done => {
-					common.stopNode('node_1');
-					setTimeout(() => {
-						done();
-					}, 2000);
+					common
+						.stopNode('node_1')
+						.then(() => {
+							done();
+						})
+						.catch(err => {
+							done(err.message);
+						});
```

Comment:
The suggested approach looks cleaner but it's more brittle; if the `stopNode()` function changes in the future, it could break stuff in an unexpected way. I think it's better to be explicit.

Reply:
If `stopNode` change in the future we should should its usage. So why not use the cleaner approach which fits in current circumstances.

Reply:
A function should be a blackbox as much as possible. The `stopNode` function shouldn't have to worry about how its output is being used by dependent functions (`done` in this case). That's why the extra fat-arrow functions are important in this case. It's also more readable because looking at the test you can see exactly what goes into the `done` function without having to dig into the code.

Reply:
There is nothing to do with internal functionality of the function. 


Yes function should be blackbox but also follow the documented interface. If its documented that the function will return a Promise then it should always return promise. And all relevant code should be written in that perspective that it is returning the Promise. 

And once function return type is change, its reference usage must be updated accordingly.

Reply:
I'll do it since most people seem to disagree with me :(